.github/workflows/rust.yml

```diff
@@ -0,0 +1,85 @@
+on: [push, pull_request]
+
+name: Rust
+
+jobs:
+  check:
+    name: Check
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        rust:
+          - stable
+          - 1.43.0 # Debian 10 & Ubuntu 18.04 (LTS) repository rustc versions
```

Comment:
```suggestion
```

Against MSRV in general, we should just stick to stable and get all the language goodies as they land.

Reply:
That's not always the best idea, sticking with the latest stable version released limits the amount of users that can use the library, and using 1.43.0 isn't a version that old. This would force any user to trust `rustup` binaries and pray that they:

1. Don't introduce bugs on the compiler, as stable isn't _stable_ all times and previously some unsound code has landed the `std` crate and while the Rust team fixes them very fast after a release it's something to keep in mind.
2. Aren't maliciously modified by a bad actor. Ideally software that _will possibly handle_ peoples money or that some organization decides to trust this code to talk to monero nodes, the most sane option would be to bootstrap the `rustc` compiler, and with the crate increasing the MSRV it would be a very long and tedious process (`mrustc` supports only Rust 1.22, and from that one to 1.43.0 is a very long road).

While Rust is very flexible and good managing dependencies and makes it a lot easier to install a binary using `cargo install` it makes anyone using this crate to stick to that method instead of using a distribution `rustc` and `cargo` packages to build a binary for distribution.

There's people still using very old distributions on servers, on dedicated hardware, on old SoCs, and systems in general that can't be easily updated.

Reply:
> This would force any user to trust rustup binaries

I don't see any reason to trust Debian/Ubuntu maintainers any more than rustup. Besides, you have to balance _security_ (or security theater?) against convenience.

> There's people still using very old distributions on servers, on dedicated hardware, on old SoCs, and systems in general that can't be easily updated.

That's too bad, but should not lead to unnecessary self-imposed limitations.

FWIW [Parity](https://substrate.dev/docs/en/knowledgebase/getting-started/#rust-developer-environment), me and most other blockchain companies develop against `stable`.

Reply:
>I don't see any reason to trust Debian/Ubuntu maintainers any more than rustup. Besides, you have to balance security (or security theater?) against convenience.

IMHO not to trust their binaries, but their infrastructure, if you wanted you could spin up your own Debian build without trusting anyone else, it already works, and if not, you could use Guix, etc, and all the distributions, and almost 100% of them aren't keeping up with the latest rustc, to keep an eye on this, the latest LTS of Ubuntu is still on [1.43](https://packages.ubuntu.com/focal/rustc)

>FWIW Parity, me and most other blockchain companies develop against stable.

Most is an overstatement, and I wont criticize how a company works but the best example I could give is `rust-bitcoin` (which I contributed back on the time, while creating this crate, and used by almost all people that use Rust with Bitcoin) still supports 1.29. See the discussion from 2019 that lasted to October, this year. [<sup>1</sup>](https://github.com/rust-bitcoin/rust-bitcoin/issues/338).

I personally, don't like to choose what operating system, and Rust version and when someone should update their operating system, just for my ease for the development of an small crate, that won't probably need the new Rust features, as this is only a crate to convert from a given representation to a binary format that Monero can understand, and probably wouldn't need a version higher than that `serde` (1.13+) supports, so 1.43 is still a pretty high number.

More than theatre, it's just a basic principle that has worked all the time on computer security, something new doesn't mean secure and faster software. And yes, it's convenient to give support to a distribution package.